### PR TITLE
Run warm up test before running default tests

### PR DESCRIPTION
**Pull request by lwronski**
In the first run of the test, the output may be polluted by staff downloading, like this:
```
Downloading https://repo1.maven.org/maven2/org/scala-lang/scala-library/maven-metadata.xml
[4410](https://github.com/VirtusLab/scala-cli/actions/runs/3523852928/jobs/5908513925#step:4:4411)
Downloaded https://repo1.maven.org/maven2/org/scala-lang/scala-library/maven-metadata.xml
[4411](https://github.com/VirtusLab/scala-cli/actions/runs/3523852928/jobs/5908513925#step:4:4412)
Downloading https://repo1.maven.org/maven2/org/scala-lang/scala3-library_3/maven-metadata.xml
[4412](https://github.com/VirtusLab/scala-cli/actions/runs/3523852928/jobs/5908513925#step:4:4413)
Downloaded https://repo1.maven.org/maven2/org/scala-lang/scala3-library_3/maven-metadata.xml
[4413](https://github.com/VirtusLab/scala-cli/actions/runs/3523852928/jobs/5908513925#step:4:4414)
Downloading https://github.com/VirtusLab/scala-cli-scala-versions/raw/main/scala-versions-v1.json
[4414](https://github.com/VirtusLab/scala-cli/actions/runs/3523852928/jobs/5908513925#step:4:4415)
Downloaded https://github.com/VirtusLab/scala-cli-scala-versions/raw/main/scala-versions-v1.json
```
to clean this up, I added a warm-up test to `DefaultTests`.

To prevent CI fail such as this [error](https://github.com/VirtusLab/scala-cli/actions/runs/3523852928/jobs/5908513925#step:4:4389).

### CODE REVIEW

File: modules/integration/src/test/scala/scala/cli/integration/WithWarmUpScalaCliSuite.scala [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package scala.cli.integration
+
+import com.eed3si9n.expecty.Expecty.expect
+
+abstract class WithWarmUpScalaCliSuite extends ScalaCliSuite {
```
Comment (Gedochao) [inline modules/integration/src/test/scala/scala/cli/integration/WithWarmUpScalaCliSuite.scala:5]: NIT I think it'd be cleaner to make it a trait that you can only mixin alongside `ScalaCliSuite`
```suggestion
trait WithWarmUpTest { _:ScalaCliSuite =>
```

File: modules/integration/src/test/scala/scala/cli/integration/WithWarmUpScalaCliSuite.scala [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  def warmUpExtraTestOptions: Seq[String]
+
+  // warm-up run that downloads compiler bridges
+  // The "Downloading compiler-bridge (from bloop?) pollute the output, and would make the first test fail.
+  lazy val warmupTest: Unit = {
+    System.err.println("Running RunTests warmup test…")
```
Comment (Gedochao) [inline modules/integration/src/test/scala/scala/cli/integration/WithWarmUpScalaCliSuite.scala]:
```suggestion
    System.err.println("Running warmup test…")
```

File: modules/integration/src/test/scala/scala/cli/integration/WithWarmUpScalaCliSuite.scala [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    warmUpTests(ignoreErrors = true)
+    System.err.println("Done running RunTests warmup test.")
```
Comment (Gedochao) [inline modules/integration/src/test/scala/scala/cli/integration/WithWarmUpScalaCliSuite.scala]:
```suggestion
    System.err.println("Done running warmup test.")
```

File: modules/integration/src/test/scala/scala/cli/integration/WithWarmUpScalaCliSuite.scala (hunk 1 above)
Comment (Gedochao) [reply modules/integration/src/test/scala/scala/cli/integration/WithWarmUpScalaCliSuite.scala:5]: nvm, seems like  we need to call `super` to rely on `munit`'s implementation of `test`, which is not allowed via a trait's self-type. 
TL;DR this wouldn't work in this case.

**Review by Gedochao — APPROVED** [review]
